flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/DataFileWriter.java

```diff
@@ -55,22 +53,19 @@ public class DataFileWriter {
 
     private final RowType keyType;
     private final RowType valueType;
-    private final BulkWriter.Factory<RowData> writerFactory;
-    private final FileStatsExtractor fileStatsExtractor;
+    private final FormatWriter.Factory<RowData> writerFactory;
     private final DataFilePathFactory pathFactory;
     private final long suggestedFileSize;
 
     private DataFileWriter(
             RowType keyType,
             RowType valueType,
-            BulkWriter.Factory<RowData> writerFactory,
-            FileStatsExtractor fileStatsExtractor,
+            FormatWriter.Factory<RowData> writerFactory,
```

Comment:
All the upper layer `DataFileWriter`, `RollingFileWriter` will just use this `FormatWriter` to write records because this FormatWriter will generate column statistics automatically in their own implementations.